### DISCUSSION TITLE
How to add a translation?

### DISCUSSION THREAD

**Discussion by BreakingLead** [logisim-evolution/logisim-evolution#778] [Q&A] [answered]

**Comment by BFH-ktt1** [comment, marked as answer]
Dear @GGmyfriendXD ,
In the path src/main/resources/resources/logisim/ you will find the language directories (for example en for english, es for spanish, etc.). Inside these directories you will find the *.properties files that contain the language specific translations. Each of these files contain:
key = description
Where the key requires to be kept as is(as this is used in the sources to fetch the language specific description) and the description in the specific language.
if you find an entry like:
"# ==> key ="
This means that a language specific description is not yet available and that logisim takes the English (default) description for this key which you can find in the same file in the "en" directory.
To add a new language, this is a bit more involved. Just let me know and I prepare the "*.properties" file for you.
Note that the property files are in UTF-8
Hope this answers your question.

**Reply by yujqiao** [reply]
Hi @BFH-ktt1, I'm interested in adding a translation of Simplified Chinese. Could you please lend some support to the initial setup of files?

**Reply by BFH-ktt1** [reply]
@rapiz1 : Thanks a lot for your initiative to add a translation of Simplified Chinese!
To do this there are two steps:

1) In the directory ```src/main/resources/resources/logisim/``` you will find a file called ```settings.properties```. This file contains the "locales" that can be used by logisim. You have to add the two letter code for simplified chinese here (when I'm not mistaken ```zh```). After this logisim will be able to be switched to simplified Chinese.
2) In the directory ```src/main/resources/resources/logisim/strings``` you will find several sub-directories that contain a file ```xx.proprties```, the default English version, and files called ```xx_yy.properties``` where ```yy``` represents the language specific code that you used in step 1). Hence you have to add in each of these directories the translation files.

Please do a push-request on the ```development``` branch and thanks again a lot already for your work and initiative!

**Reply by maehne** [reply]
@rapiz1: For the record, @ManiaciaChao contributed a [Simplified Chinese localisation to Logisim ITA](https://github.com/Logisim-Ita/Logisim/pull/14) last year. Maybe, this can serve as a starting point. Note however that Logisim-evolution and Logisim ITA code bases have diverged considerably over time and it is not guaranteed that the localisation keys are the same. Also, there will be additional strings to translate for Logisim-evolution.

**Reply by yujqiao** [reply]
@maehne: Thanks for the information. I didn't know there's already a translation, though for another fork. What is even more interesting is that the contributor you mentioned and I actually know each other and we study at the same school :)

**Comment by MarcinOrlowski** [comment]
@maehne could that be moved to Discussions as well?

**Comment by maehne** [comment]
@MarcinOrlowski: Sure, that makes sense.

**Comment by R3dst0ne** [comment]
@BFH-ktt1 updating your answer to the current structure 
might be a good idea.

**Reply by MarcinOrlowski** [reply]
We should add this to official docs

**Reply by mbaillif** [reply]
I'll look into publishing something on this.

**Comment by MarcinOrlowski** [comment]
I added the above to project docs in #1171 so this thread can be removed after next release (when docs are `master)
